pkg/kademlia/workers_test.go

```diff
@@ -0,0 +1,213 @@
+	"google.golang.org/grpc"
+	"storj.io/storj/internal/mock"
+	"storj.io/storj/pkg/dht/mocks"
+	"storj.io/storj/pkg/node"
+	"storj.io/storj/pkg/provider"
+	proto "storj.io/storj/protos/overlay"
+)
+
+var (
+	ctx = context.Background()
+)
+
+func TestGetWork(t *testing.T) {
+	cases := []struct {
+		name     string
+		worker   *worker
+		expected *proto.Node
+		ch       chan *proto.Node
+	}{
+		{
+			name:     "test valid chore returned",
+			worker:   newWorker(context.Background(), nil, []*proto.Node{&proto.Node{Id: "1001"}}, nil, node.StringToID("1000"), 5),
+			expected: &proto.Node{Id: "1001"},
+			ch:       make(chan *proto.Node, 2),
+		},
+		{
+			name: "test no chore left",
+			worker: func() *worker {
+				w := newWorker(context.Background(), nil, []*proto.Node{&proto.Node{Id: "foo"}}, nil, node.StringToID("foo"), 5)
+				w.maxResponse = 0
+				w.pq.Pop()
+				assert.Len(t, w.pq, 0)
+				return w
+			}(),
+			expected: nil,
+			ch:       make(chan *proto.Node, 2),
+		},
+	}
+
+	for _, v := range cases {
+		ctx, cf := context.WithTimeout(context.Background(), 50*time.Millisecond)
+		defer cf()
+
+		v.worker.cancel = cf
+		v.worker.getWork(ctx, v.ch)
+
+		if v.expected != nil {
+			actual := <-v.ch
+			assert.Equal(t, v.expected, actual)
+		} else {
+			assert.Len(t, v.ch, 0)
+		}
+
+	}
+}
+
+func TestWorkerLookup(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockDHT := mock_dht.NewMockDHT(ctrl)
+	mockRT := mock_dht.NewMockRoutingTable(ctrl)
+
+	lis, err := net.Listen("tcp", ":0")
+	assert.NoError(t, err)
+
+	srv, mock := newTestServer(nil)
+	go func() { _ = srv.Serve(lis) }()
+	defer srv.Stop()
+	cases := []struct {
+		name     string
+		worker   *worker
+		work     *proto.Node
+		expected []*proto.Node
+	}{
+		{
+			name: "test valid chore returned",
+			worker: func() *worker {
+				ca, err := provider.NewCA(ctx, 12, 4)
+				assert.NoError(t, err)
+				identity, err := ca.NewIdentity()
+				assert.NoError(t, err)
+				nc, err := node.NewNodeClient(identity, proto.Node{Id: "foo", Address: &proto.NodeAddress{Address: ":0"}}, mockDHT)
+				assert.NoError(t, err)
+				mock.returnValue = []*proto.Node{&proto.Node{Id: "foo"}}
+				return newWorker(context.Background(), nil, []*proto.Node{&proto.Node{Id: "foo"}}, nc, node.StringToID("foo"), 5)
+			}(),
+			work:     &proto.Node{Id: "foo", Address: &proto.NodeAddress{Address: lis.Addr().String()}},
+			expected: []*proto.Node{&proto.Node{Id: "foo"}},
+		},
+	}
+
+	for _, v := range cases {
+		mockDHT.EXPECT().GetRoutingTable(gomock.Any()).Return(mockRT, nil)
+		mockRT.EXPECT().ConnectionSuccess(gomock.Any()).Return(nil)
+		actual := v.worker.lookup(context.Background(), v.work)
+		assert.Equal(t, v.expected, actual)
+		assert.Equal(t, int32(1), mock.queryCalled)
+	}
+}
+
+func TestUpdate(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockDHT := mock_dht.NewMockDHT(ctrl)
+
+	lis, err := net.Listen("tcp", ":0")
```

Comment:
`127.0.0.1:0`

Reply:
why?

Reply:
I agree with @coyle, we should test for both IPv4 and IPv6